idaes/generic_models/unit_models/pressure_changer.py

```diff
@@ -455,6 +479,38 @@ def actual_work(b, t):
                     b.work_isentropic[t] * b.efficiency_isentropic[t]
                 )
 
+        # Add head expressions (often calcuated from performance curve)
+        try:
+            @self.Expression(self.flowsheet().config.time)
+            def head_isentropic(b, t): # units are energy/mass e.g. J/kg
+                if hasattr(b.control_volume.properties_in[t], "flow_mass"):
+                    return (b.work_isentropic[t] /
+                        b.control_volume.properties_in[t].flow_mass)
+                else:
+                    return (b.work_isentropic[t] /
+                        b.control_volume.properties_in[t].flow_mol /
+                        b.control_volume.properties_in[t].mw)
+
+            @self.Expression(self.flowsheet().config.time)
+            def head(b, t): # units are energy/mass e.g. J/kg
+                if hasattr(b.control_volume.properties_in[t], "flow_mass"):
+                    return (b.work_mechanical[t] /
+                        b.control_volume.properties_in[t].flow_mass)
+                else:
+                    return (b.work_mechanical[t] /
+                        b.control_volume.properties_in[t].flow_mol /
+                        b.control_volume.properties_in[t].mw)
+
+        except PropertyNotSupportedError:
+            _log.warning(
+                "flow_mass or flow_mol and mw not supported by property package"
+                " but are required for isentropic pressure changer head"
+                " calculation")
+
+        # Add performace curve block.  If this block is empty that's fine
```

Comment:
Should we put this in an `if` statement which checks to see if the user provides a performance curve object? If they don't we probably don't need to build this block.

Reply:
It's set up so you can just add constraints to the performance_curve block later without the callback.  I thought the option may seem more straight forward to a lot of people, even-though it's not much different.

Reply:
Also, just noted a typo: "performace"